bigquery/snippets/managedwriter/bigquerystorage_append_rows_default.go

```diff
@@ -0,0 +1,199 @@
+		sList := make([]*exampleproto.SampleStruct, 5)
+		for i := 0; i < int(random.Int63n(5)+1); i++ {
+			sList[i] = &exampleproto.SampleStruct{
+				SubIntCol: proto.Int64(random.Int63()),
+			}
+		}
+
+		m := &exampleproto.SampleData{
+			BoolCol:    proto.Bool(true),
+			BytesCol:   []byte("some bytes"),
+			Float64Col: proto.Float64(3.14),
+			Int64Col:   proto.Int64(123),
+			StringCol:  proto.String("example string value"),
+
+			// These types require special encoding/formatting to transmit.
+
+			// DATE values are number of days since the Unix epoch.
+
+			DateCol: proto.Int32(int32(time.Now().UnixNano() / 86400000000000)),
+
+			// DATETIME uses the literal format.
+			DatetimeCol: proto.String("2022-01-01 12:13:14.000000"),
+
+			// GEOGRAPHY uses Well-Known-Text (WKT) format.
+			GeographyCol: proto.String("POINT(-122.350220 47.649154)"),
+
+			// NUMERIC and BIGNUMERIC can be passed as string, or more efficiently
+			// using a packed byte representation.
+			NumericCol:    proto.String("99999999999999999999999999999.999999999"),
+			BignumericCol: proto.String("578960446186580977117854925043439539266.34992332820282019728792003956564819967"),
+
+			// TIME also uses literal format.
+			TimeCol: proto.String("12:13:14.000000"),
+
+			// TIMESTAMP uses microseconds since Unix epoch.
+			TimestampCol: proto.Int64(time.Now().UnixNano() / 1000),
+
+			// Int64List is an array of INT64 types.
+			Int64List: []int64{2, 4, 6, 8},
+
+			// This is a required field, and thus must be present.
+			RowNum: proto.Int64(23),
+
+			// StructCol is a single nested message.
+			StructCol: &exampleproto.SampleStruct{
+				SubIntCol: proto.Int64(random.Int63()),
+			},
+
+			// StructList is a repeated array of a nested message.
+			StructList: sList,
+		}
+
+		b, err := proto.Marshal(m)
+		if err != nil {
+			return nil, fmt.Errorf("error generating message %d: %w", i, err)
+		}
+		msgs[i] = b
+	}
+	return msgs, nil
+}
+
+// appendToDefaultStream demonstrates using the managedwriter package to write some example data
+// to a default stream.
+func appendToDefaultStream(w io.Writer, projectID, datasetID, tableID string) error {
+	// projectID := "myproject"
+	// datasetID := "mydataset"
+	// tableID := "mytable"
+
+	ctx := context.Background()
+	// Instantiate a managedwriter client to handle interactions with the service.
+	client, err := managedwriter.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("managedwriter.NewClient: %w", err)
+	}
+	// Close the client when we exit the function.
+	defer client.Close()
+
+	// We need to communicate the descriptor of the protocol buffer message we're using, which
+	// is analagous to the "schema" for the message.  Both SampleData and SampleStruct are
+	// two distinct messages in the compiled proto file, so we'll use adapt.NormalizeDescriptor
+	// to unify them into a single self-contained descriptor representation.
+	m := &exampleproto.SampleData{}
```

Comment:
nit: you can also just declare this variable with `var`, which has the effect of initializing it to its nil value, for example:

```
var m *exampleproto.SampleData
```

Reply:
switched